cli/azd/pkg/oneauth/utils.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+//go:build oneauth
+
+package oneauth
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync/atomic"
+)
+
+const applicationID = "com.microsoft.azd"
```

Comment:
Do we need to register this anywhere?  Also, in other cases we've re-used the App ID of the `az` CLI (so, for example, the app is pre-consented for a variety of operations).  Do we need to do something here for that or is this just about segmenting storage inside the OneAuth library?

Reply:
OneAuth uses this as a key for account management. Practically speaking, at this point it's an arbitrary value that just needs to be unique. I suppose OneAuth's telemetry system could use it for correlation, in which case I imagine it would be registered as part of onboarding to that system but I haven't looked into that process and don't have azd sending telemetry anyway.